packages/material-ui/src/ButtonGroup/ButtonGroup.test.js

```diff
@@ -0,0 +1,229 @@
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedOutlined), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedOutlinedPrimary), false);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedOutlinedSecondary), true);
+  });
+
+  it('should render a contained button', () => {
+    const wrapper = shallow(
+      <ButtonGroup variant="contained">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().variant, 'contained');
+    assert.strictEqual(wrapper.childAt(0).props().color, 'default');
+    assert.strictEqual(wrapper.hasClass(classes.contained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.grouped), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedPrimary), false);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedSecondary), false);
+  });
+
+  it('should render a contained primary button', () => {
+    const wrapper = shallow(
+      <ButtonGroup variant="contained" color="primary">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().variant, 'contained');
+    assert.strictEqual(wrapper.childAt(0).props().color, 'primary');
+    assert.strictEqual(wrapper.hasClass(classes.contained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.grouped), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedPrimary), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedSecondary), false);
+  });
+
+  it('should render a contained secondary button', () => {
+    const wrapper = shallow(
+      <ButtonGroup variant="contained" color="secondary">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().variant, 'contained');
+    assert.strictEqual(wrapper.childAt(0).props().color, 'secondary');
+    assert.strictEqual(wrapper.hasClass(classes.contained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.grouped), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContained), true);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedPrimary), false);
+    assert.strictEqual(wrapper.childAt(0).hasClass(classes.groupedContainedSecondary), true);
+  });
+
+  it('should render a small button', () => {
+    const wrapper = shallow(
+      <ButtonGroup size="small">
+        <Button>Hello World</Button>
+      </ButtonGroup>,
+    );
+    assert.strictEqual(wrapper.childAt(0).props().size, 'small');
+  });
+
+  it('should render a large button', () => {
```

Comment:
We should (no pun intended) communicate these differently. There's no difference in the description between default behavior and controlled behavior.

default:
> it should render a medium button

controlled:
> it can render a large button